File: src/client/test/app/SparklineTest.js
/**
 * A test class for the echopoint.google.chart.Sparkline client-side component.
 * Displays simple and multi value charts to test Google Chart API interaction.
 *
 * @author Rakesh 2008-08-20
 * @version $Id: SparklineTest.js,v 1.1.1.1 2010-04-01 09:47:33 perxi Exp $
 */
echopoint.test.SparklineTest = Core.extend(
{
  $construct: function( testArea )
  {
    testArea.add( this._createSimple() );
    testArea.add( this._createComplex() );
  },

  _createSimple: function()
  {
    var xdata = [ 30,60,70,90,95,110 ];
    var xmax = 120;

    var markers = new Array();
    markers[0] = new echopoint.google.chart.model.ShapeMarker( "o", "ff3333", 5 );

    var data = new echopoint.google.chart.model.ChartData( xdata, xmax );
    data.color = "00ff00";
    data.markers = markers;

    var title = new echopoint.google.chart.model.Title();
    title.add( "Simple Sparkline" );

    return new echopoint.google.chart.Sparkline(
    {
      renderId: "echopointUnitTestSimpleSparkline",
      styleName: "SimpleChart",
      data: [ data ],
      title: title
    });
  },

  _createComplex: function()
  {
    var markers1 = new Array();
    markers1[0] = new echopoint.google.chart.model.ShapeMarker( "a", "00ff33", 7 );
    markers1[1] = new echopoint.google.chart.model.ShapeMarker( "c", "00ff33", 7 );
    markers1[2] = new echopoint.google.chart.model.ShapeMarker( "d", "00ff33", 7 );
    markers1[3] = new echopoint.google.chart.model.ShapeMarker( "o", "00ff33", 7 );
    markers1[4] = new echopoint.google.chart.model.ShapeMarker( "s", "00ff33", 7 );
    markers1[5] = new echopoint.google.chart.model.ShapeMarker( "tValue+7", "00ff33", 7 );
    markers1[6] = new echopoint.google.chart.model.ShapeMarker( "x", "00ff33", 7 );

    var data = new Array();
    data[0] = new echopoint.google.chart.model.ChartData(
        [ 0, 30, 60, 70, 90, 95, 100 ], 110 );
    data[0].ydata = [ 20,30,40,50,60,70,80 ];
    data[0].ymax = 100;
    data[0].legend = "First";
    data[0].markers = markers1;

    var markers2 = new Array();
    markers2[0] = new echopoint.google.chart.model.ShapeMarker( "x", "ff00ff", 7 );
    markers2[1] = new echopoint.google.chart.model.ShapeMarker( "s", "ff00ff", 7 );
    markers2[2] = new echopoint.google.chart.model.ShapeMarker( "o", "ff00ff", 7 );
    markers2[3] = new echopoint.google.chart.model.ShapeMarker( "d", "ff00ff", 7 );
    markers2[4] = new echopoint.google.chart.model.ShapeMarker( "c", "ff00ff", 7 );
    markers2[5] = new echopoint.google.chart.model.ShapeMarker( "a", "ff00ff", 7 );

    data[1] = new echopoint.google.chart.model.ChartData(
        [ 10,30,40,45,52 ], 60 );
    data[1].ydata = [ 100,90,40,20,10 ]
    data[1].ymax = 110;
    data[1].legend = "Second";
    data[1].markers = markers2;

    var markers3 = new Array();
    markers3[0] = new echopoint.google.chart.model.ShapeMarker( "x", "ff3300", 7 );
    markers3[1] = new echopoint.google.chart.model.ShapeMarker( "a", "ff3300", 7 );
    markers3[2] = new echopoint.google.chart.model.ShapeMarker( "o", "ff3300", 7 );
    markers3[3] = new echopoint.google.chart.model.ShapeMarker( "c", "ff3300", 7 );
    markers3[4] = new echopoint.google.chart.model.ShapeMarker( "t", "ff3300", 7 );

    data[2] = new echopoint.google.chart.model.ChartData( [ -1 ], 0 );
    data[2].ydata = [ 5,33,50,55,7 ];
    data[2].ymax = 60;
    data[2].legend = "Third";
    data[2].markers = markers3;

    var title = new echopoint.google.chart.model.Title( "Complex Sparkline" );
    title.add( "Multiple Line Title" );

    var lineStyles = new Array();
    lineStyles[0] = new echopoint.google.chart.model.LineStyle( 3, 6, 3 );
    lineStyles[1] = new echopoint.google.chart.model.LineStyle( 2, 4, 2 );
    lineStyles[2] = new echopoint.google.chart.model.LineStyle( 4 );

    return new echopoint.google.chart.Sparkline(
    {
      renderId: "echopointUnitTestStrutComplexSparkline",
      styleName: "ComplexChart",
      data: data,
      title: title,
      lineStyles: lineStyles
    });
  }
});

File: src/server-java/test/echopoint/SparklineTest.java
/*
 * This file is part of the Echo Point Project.  This project is a
 * collection of Components that have extended the Echo Web Application
 * Framework Version 3.
 *
 * Version: MPL 1.1
 *
 * The contents of this file are subject to the Mozilla Public License Version
 * 1.1 (the "License"); you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 * http://www.mozilla.org/MPL/
 *
 * Software distributed under the License is distributed on an "AS IS" basis,
 * WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License
 * for the specific language governing rights and limitations under the
 * License.
 */

package echopoint;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertNotNull;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collection;
import java.util.List;

import nextapp.echo.app.Component;

import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;

import echopoint.google.chart.Sparkline;
import echopoint.google.chart.model.ChartData;
import echopoint.google.chart.model.LineStyle;
import echopoint.google.chart.model.ShapeMarker;
import echopoint.google.chart.model.Title;

/**
 * Unit test suite for the {@link echopoint.google.chart.Sparkline} component
 * wrapper.
 *
 * @author Rakesh Vidyadharan 2008-08-20
 * @version $Id: SparklineTest.java,v 1.1.1.1 2010-04-01 09:47:30 perxi Exp $
 */
public class SparklineTest extends GoogleChartTest<Sparkline<Integer>>
{
  @BeforeClass
  public static void init()
  {
    set( new Sparkline<Integer>() );
    setData( new ChartData<Integer>() );
  }

  @Test
  public void renderId()
  {
    final String id = "echopointUnitTestSimpleSparkline";
    getComponent().setRenderId( id );
    assertEquals( "Ensuring render id is same", id, getComponent().getRenderId() );
  }

  @Test
  public void data()
  {
    final Integer[] array = new Integer[] { 30,60,70,90,95,110 };
    final List<Integer> xdata = Arrays.asList( array );
    final int xmax = 120;

    getData().setXdata( xdata );
    getData().setXmax( xmax );
  }

  @Test
  public void markers()
  {
    final ArrayList<ShapeMarker> markers = new ArrayList<ShapeMarker>();
    markers.add( new ShapeMarker( "o", "ff3333", 5 ) );
    getData().setMarkers( markers );

    assertFalse( "Ensuring markers set", getData().getMarkers().isEmpty() );
  }

  @Test
  public void color()
  {
    final String color = "00ff00";
    getData().setColor( color );
    assertEquals( "Ensuring color set", color, getData().getColor() );
  }

  @Test
  public void title()
  {
    final Title title = new Title();
    title.add( "Sparkline" );
    getComponent().setTitle( title );

    assertEquals( "Ensuring title set", title, getComponent().getTitle() );
  }

  @Test
  public void lineStyles()
  {
    final Collection<LineStyle> styles = new ArrayList<LineStyle>();
    styles.add( new LineStyle( 3, 6, 3 ) );

    getComponent().setLineStyles( styles );
    assertNotNull( "Ensuring line styles set", getComponent().getLineStyles() );
  }

  @AfterClass
  @SuppressWarnings( {"unchecked"} )
  public static void finish()
  {
    final Component content = Application.getContent().getTestArea();
    content.removeAll();

    final Sparkline<Integer> chart = (Sparkline<Integer>) get();
    final ArrayList<ChartData<Integer>> collection = new ArrayList<ChartData<Integer>>();
    collection.add( getData() );
    chart.setData( collection );
    assertNotNull( "Ensuring that data is set", chart.getData() );

    content.add( chart );
  }
}